starport/interface/cli/starport/main.go

```diff
@@ -37,17 +37,15 @@ func main() {
 	ctx := clictx.From(context.Background())
 	err := starportcmd.New().ExecuteContext(ctx)
 
-	addMetric(Metric{
-		Err: err,
-	})
-
 	if err == context.Canceled {
+		addMetric(Metric{
+			Err: err,
+		})
 		fmt.Println("aborted")
 		return
 	}
 	if err != nil {
 		fmt.Println()
-		fmt.Println(err)
-		os.Exit(1)
+		panic(err)
```

Comment:
Linteris complaining when using `exit` with a deferred function.
I didn't see the difference since the panic is recovered and the error printed. Otherwise, we can still silent this issue since the defer is only about recover

Reply:
ah, okay. it should be fine then.